### ISSUE TITLE
FIx Locate button in sidenav

### ISSUE THREAD

**Issue by yexingsha** [zotero/zotero#4253]
The menu needs to use context menu behavior to prevent it from cutting into the side of the screen:

<img width="218" alt="Screenshot 2024-06-18 at 01 03 15" src="https://github.com/zotero/zotero/assets/57771795/f81105b3-65c5-401c-8bcb-08fc8d2ab50d">

And it shouldn't have the same "Pin Section" context menu as other sidenav buttons:

<img width="127" alt="Screenshot 2024-06-18 at 01 04 27" src="https://github.com/zotero/zotero/assets/57771795/e1d13fe5-a33d-4017-a3cd-746291fc3318">

**Comment by ramonmi** [comment]
Similarly, the Notes in the side pane of the reader shouldn't have the "Pin Section" context menu. This button has no effect.

**Comment by yexingsha** [comment]
Yeah I was thinking of the second option, which is the behavior we use for item info buttons, but covering the button is less acceptable in this case. The thing that bothers me a little is that native menus like the one in finder always maintain 6px spacing from the edge of the screen, whereas our menus extend a little outside the screen. @windingwind has [offered some insights into this](https://github.com/zotero/zotero/pull/4200#issuecomment-2150375718), and it's a firefox issue that we can't do anything about? So let's just keep it as is.
